### PR TITLE
[Soundcloud] Also use avatar picture for thumbnail (fixes issue #12878)

### PR DESCRIPTION
**Pull request by parmjitv**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

Updated the extractor for the Soundcloud.com website to use the avatar picture when the album artwork is not available - issue #12878.  Apparently a number of tracks on this site do not feature any album art.

Cheers,

Parmjit V.

### CODE REVIEW

**Comment by parmjitv** [comment]
@dstftw - Are any collaborators available for a first review of this PR?  I believe the fix for this issue is pretty straightforward.  Much appreciated!

**Comment by parmjitv** [comment]
@dstftw - Does this PR require any corrections? Thank you!

**Comment by yan12125** [comment]
The code looks fine. Just need a test case in _TESTS.

**Comment by parmjitv** [comment]
@yan12125 - The test suite has been updated as requested.  Please let me know if there are any other corrections that need to be made.  Cheers!

File: youtube_dl/extractor/soundcloud.py [hunk 1]
```diff
@@ -136,6 +136,38 @@ class SoundcloudIE(InfoExtractor):
...
+                'ext': 'mp3',
+                'title': 'rockstar (feat. 21 Savage)',
+                'description': None,
+                'uploader': 'Post Malone',
+                'upload_date': '20170908',
+                'duration': 218,
+                'thumbnail': 'https://i1.sndcdn.com/artworks-JbxEN7x8VfRc-0-t500x500.jpg',
+                'license': 'all-rights-reserved',
+            },
+        },
```
Comment (dstftw) [inline youtube_dl/extractor/soundcloud.py]: Remove this test.

File: youtube_dl/extractor/soundcloud.py [hunk 2]
```diff
@@ -136,6 +136,38 @@ class SoundcloudIE(InfoExtractor):
...
+            'md5': '59c7872bc44e5d99b7211891664760c2',
+            'info_dict': {
+                'id': '309699954',
+                'ext': 'mp3',
+                'title': 'Sideways (Prod. Mad Real)',
+                'description': 'md5:d41d8cd98f00b204e9800998ecf8427e',
+                'uploader': 'garyvee',
+                'upload_date': '20170226',
+                'duration': 207,
+                'thumbnail': 'https://i1.sndcdn.com/avatars-000298106657-ogrgmg-t500x500.jpg',
```
Comment (dstftw) [inline youtube_dl/extractor/soundcloud.py]: Do not use exact URLs.

File: youtube_dl/extractor/soundcloud.py [hunk 3]
```diff
@@ -136,6 +136,38 @@ class SoundcloudIE(InfoExtractor):
...
+                'license': 'all-rights-reserved',
+            },
```
Comment (dstftw) [inline youtube_dl/extractor/soundcloud.py]: Skip download.

**Comment by parmjitv** [comment]
@dstftw - Done, thanks for the prompt feedback.  Please let me know if anything else is still outstanding for this PR.  Cheers!

**Pull request merged**